bundles/org.openhab.binding.intesis/src/main/java/org/openhab/binding/intesis/internal/handler/IntesisHomeHandler.java

```diff
@@ -0,0 +1,557 @@
+                Id id = gson.fromJson(data.id.toString(), Id.class);
+                if (id.sessionID.toString() != null && !id.sessionID.toString().isEmpty()) {
+                    sessionId = id.sessionID.toString();
+                    updateStatus(ThingStatus.ONLINE);
+                } else {
+                    updateStatus(ThingStatus.OFFLINE);
+                }
+            }
+        }
+    }
+
+    public @Nullable String logout() {
+        String contentString = "{\"command\":\"logout\",\"data\":{\"sessionID\":\"" + sessionId + "\"}}";
+        String response = api.postRequest(ipAddress, contentString);
+        return response;
+    }
+
+    public boolean getSuccess(String response) {
+        Response resp = gson.fromJson(response, Response.class);
+        boolean success = resp.success;
+        return success;
+    }
+
+    public void getInfo() {
+        String contentString = "{\"command\":\"getinfo\",\"data\":\"\"}";
+        String response = api.postRequest(ipAddress, contentString);
+        logger.trace("getInfo response : {}", response);
+        if (response != null) {
+            boolean success = getSuccess(response);
+            if (success) {
+                Response resp = gson.fromJson(response, Response.class);
+                Data data = gson.fromJson(resp.data.toString(), Data.class);
+                Info info = gson.fromJson(data.info.toString(), Info.class);
+                properties.put(PROPERTY_VENDOR, "Intesis");
+                properties.put(PROPERTY_MODEL_ID, info.deviceModel);
+                properties.put(PROPERTY_SERIAL_NUMBER, info.sn);
+                properties.put(PROPERTY_FIRMWARE_VERSION, info.fwVersion);
+                properties.put(PROPERTY_MAC_ADDRESS, info.wlanSTAMAC);
+                updateProperties(properties);
+                updateStatus(ThingStatus.ONLINE);
+            }
+        }
+    }
+
+    public void getDataPoints() {
+        login();
+        if (!sessionId.isEmpty()) {
+            String contentString = "{\"command\":\"getavailabledatapoints\",\"data\":{\"sessionID\":\"" + sessionId
+                    + "\"}}";
+            String response = api.postRequest(ipAddress, contentString);
+            logger.trace("available Datapoints response : {}", response);
+            if (response != null) {
+                boolean success = getSuccess(response);
+                if (success) {
+                    logout();
+                    Response resp = gson.fromJson(response, Response.class);
+                    Data data = gson.fromJson(resp.data.toString(), Data.class);
+                    Dp dp = gson.fromJson(data.dp.toString(), Dp.class);
+                    Datapoints[] datapoints = gson.fromJson(dp.datapoints, Datapoints[].class);
+                    for (int i = 0; i < datapoints.length; i++) {
+                        Descr descr = gson.fromJson(datapoints[i].descr, Descr.class);
```

Comment:
A for-each loop would be preferred here unless you really need to know the current index.
```suggestion
                    for (Datapoints datapoint : datapoints) {
                        Descr descr = gson.fromJson(datapoint.descr, Descr.class);
```

Reply:
Done